main.rb

```diff
@@ -1,44 +1,54 @@
 require_relative 'app'
+require_relative 'user_input_helper'
 
-def render_choices
-  puts 'Choose your Option:'
-  puts '1. List all books.'
-  puts '2. List all people.'
-  puts '3. Create a person.'
-  puts '4. Create a book.'
-  puts '5. Create a rental.'
-  puts '6. List all rentals for a given person id.'
-  puts '7. Quit'
-end
+class Main
+  includes UserInputHelper
 
-def choose_a_number(choice, app)
-  case choice
-  when 1
-    app.list_books
-  when 2
-    app.list_people
-  when 3
-    app.create_a_person
-  when 4
-    app.create_a_book
-  when 5
-    app.create_a_rental
-  when 6
-    app.list_rentals
-  else
-    puts '👋 Sorry to see you leaving..'
+  def initialize
+    @app = App.new
   end
-end
 
-def main
-  app = App.new
-  loop do
-    render_choices
-    choice = gets.chomp.to_i
-    break if choice == 7
+  def start
+    loop do
+      render_choices
+      choice = gets.chomp.to_i
+      break if choice == 7
+
+      choose_a_number(choice)
+    end
+  end
+
+  private
+
+  def render_choices
+    puts 'Choose your Option:'
+    puts '1. List all books.'
+    puts '2. List all people.'
+    puts '3. Create a person.'
+    puts '4. Create a book.'
+    puts '5. Create a rental.'
+    puts '6. List all rentals for a given person id.'
+    puts '7. Quit'
+  end
 
-    choose_a_number(choice, app)
+  def choose_a_number(choice)
+    case choice
+    when 1
+      @app.list_books
+    when 2
+      @app.list_people
+    when 3
+      @app.create_a_person
+    when 4
+      @app.create_a_book
```

Comment:
- [ ] Having made great effort refactoring your code in line with the previous reviewer requirement, I notice i found it difficult creating a book and whenever i tried to do that, it throws an error and automatically stops the app. I would kindly suggest you work on that and the sure to have the app running and working perfectly as suggested by the previous reviewer.

Screenshot 👇 

![person](https://github.com/ANTOINE1128/OOP--School--Library/assets/78634841/707e3445-8c75-46b6-a068-822a9bb3646d)